src/imageOperations/extractedImage.ts

```diff
@@ -0,0 +1,17 @@
+import { Buffer } from "node:buffer";
+import { BufferInput } from "../input/sources";
+
+
+export class ExtractedImage {
+  readonly pageId: number;
+  imageData: Buffer;
+
+  constructor(imageData: Uint8Array, pageId: number) {
+    this.pageId = pageId;
+    this.imageData = Buffer.from(imageData);
+  }
+
+  asSource(): BufferInput {
+    return new BufferInput({ buffer: this.imageData, filename: `receipt_p${this.pageId}.pdf` });
```

Comment:
the filename should not include the word 'receipt' in it since this is a generic class

also, what happens when there are multiple images on the same page?

Reply:
hmm, thinking this could be an abstract class or even an interface, WDYT?